### DISCUSSION TITLE
Help with my Uni Project

### DISCUSSION THREAD

**Discussion by MiniMenzu** [firefly-iii/firefly-iii#9384] [General]
Hii, My name is Menura, and I am an undergraduate student currently working on an AI based personal finance management tool for my university project. The project aims to help users track and manage their expenses more effectively, using features like automated transaction categorization, budget recommendations, and spending insights.

I came to find firefly iii by coincidence and it caught my interest. I would be grateful if you could provide me with some tips or guidance on how to approach this project, especially in terms of starting points, key features to focus on, and any challenges I should be aware of.

Your insights would be incredibly helpful in ensuring the success of my project.

Thank you for your time, and I look forward to your response.

Best regards,
Menura

**Comment by JC5** [comment]
Hey, thanks for starting a discussion.

I'm not entirely sure what you would like to approach? Are you looking for a development point of view, or what it would be like as a new user? Do you need installation guidance?

**Reply by MiniMenzu** [reply]
I'm a beginner in Python and machine learning. I would like to know how to start developing a project like this. If you could provide some guidance on how to approach the project, it would be very helpful.

**Reply by JC5** [reply]
That's a big question, lol. I can't really answer that. I have a start though: https://docs.firefly-iii.org/explanation/more-information/architecture/

First, you must develop something of a model. How do transactions, accounts and budgets relate? Then, try creating a database that reflects your model. Then you build or boilerplate forms and lists etc. And then you can start working on charts.

**Comment by MiniMenzu** [comment]
Thank you for that info. What's the basic coding  language used in the Laravel? Because I was thinking about doing it using python

**Reply by JC5** [reply]
Firefly III is written in PHP 8.3.

**Reply by MiniMenzu** [reply]
Is laravel compatible with python

**Reply by JC5** [reply]
Sorry, this is something that must google yourself.
